nf1_cellprofiler_classifier_fs_all_cellprofiler_all_feat_perfect.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Feature Analysis and Genotype Classification
+
+# ## Imports
+
+# In[1]:
+
+
+import pandas as pd
+from pandas.plotting import scatter_matrix
+import numpy as np
+import pathlib
+from sklearn.preprocessing import LabelBinarizer, StandardScaler
+from sklearn.linear_model import LogisticRegression
+from sklearn.ensemble import AdaBoostClassifier
+from xgboost import XGBClassifier
+from sklearn.model_selection import KFold, train_test_split
+from sklearn.metrics import accuracy_score, confusion_matrix, ConfusionMatrixDisplay
+from sklearn.decomposition import PCA
+import matplotlib.pyplot as plt
+import umap
+
+
+# # Seed and Generator for Reproducibility
+
+# In[2]:
+
+
+rnd_val = 0 # Random value for all seeds
+rng = np.random.default_rng(seed=rnd_val) # random number generator
+
+
+# # Converting csv to pandas dataframe
+
+# In[3]:
+
+
+morph_features = 'nf1_sc_norm_fs_all_cellprofiler.csv.gz'
+
+
+# In[4]:
+
+
+data_dir = pathlib.Path("nf1_data_repo/4_processing_features/data/Plate1/CellProfiler")
+data_comp = data_dir / morph_features
+
+
+# In[5]:
+
+
+# If the file isn't found in the path above then raise an error.
+if not data_comp.is_file():
+    raise FileNotFoundError(f"File '{data_comp}' does not exist")
+    
+morph_feat = pd.read_csv(data_comp)
+
+
+# # Creating binary labels for the genotype
+
+# In[6]:
+
+
+lb = LabelBinarizer()
+morph_feat['genotype_label'] = lb.fit_transform(morph_feat['Metadata_genotype'])
+featdf = morph_feat # Create a copy for modification
+
+
+# # Selecting Features
+
+# In[7]:
+
+
+feat_col = [col for col in featdf.columns if 'Metadata' not in col] # Select all columns that don't contain the Metadata in their name
+
+featdf = featdf[feat_col]
+featdf = featdf.loc[:,featdf.columns != 'Unnamed: 0'] # Remove the unamed column
+
+
+# # Sampling
+
+# In[8]:
+
+
+min_class_samps_size = min(featdf['genotype_label'].value_counts().values) # Sample size for the minority class
+dsamp_featdf = featdf.groupby('genotype_label', group_keys=False).apply(lambda x: x.sample(n=min_class_samps_size, random_state=rnd_val)) # Down-sample features according to minority class
+traindf, testdf = train_test_split(dsamp_featdf, random_state=rnd_val, shuffle=True, train_size=0.9) # Train and Test Dataframes
```

Comment:
```suggestion
traindf, _ , testdf, _ = train_test_split(dsamp_featdf, random_state=rnd_val, shuffle=True, train_size=0.9) # Train and Test Dataframes
```
I may be wrong, but I think you're only storing the X_train and X_test set in your varaibles. 

According to the documentation, the  [`train_test_splt`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) returns 4 variables `X_train, X_test, y_train, y_test  `. If your intention is not to use certain variables, I recommend using `_` as the "unused" variable

Reply:
Adding to what Erik mentions, I also felt concerned about this and wonder if it may change the results.

Reply:
That makes sense, ya, it works either way.